src/processor/reconnection-graph.service.ts

```diff
@@ -435,26 +439,94 @@ export class ReconnectionGraphService {
+    dsnpUserId: MessageSourceId,
+    providerId: ProviderId,
+    providerKeys: KeyringPair,
+    batchesMap: SubmittableExtrinsic<'rxjs', ISubmittableResult>[][],
+    ): Promise<void> {
+    try {
+      // iterate over batches and send them to the chain
+      let batchPromises: Promise<void>[] = [];
+
+      batchesMap.forEach(async (batch) => {
+        batchPromises.push(this.processSingleBatch(dsnpUserId, providerId, providerKeys, batch));
+      });
+
+      await Promise.all(batchPromises);
+
+    } catch (e) {
+      this.logger.error(`Error processing batches for ${dsnpUserId.toString()}: ${e}`);
+      throw e;
+    }
+  }
+
+  async processSingleBatch(
+    dsnpUserId: MessageSourceId,
+    provideId: ProviderId,
+    providerKeys: KeyringPair,
+    batch: SubmittableExtrinsic<'rxjs', ISubmittableResult>[]
+    ): Promise<void> {
+    try {
+      const [event, eventMap] = await this.blockchainService.createExtrinsic(
+        { pallet: 'frequencyTxPayment', extrinsic: 'payWithCapacityBatchAll' }, 
+        { eventPallet: 'utility', event: 'BatchCompleted' },
+        providerKeys, batch).signAndSend();
+      
       if (!event) {
-        throw new Error(`Error submitting extrinsic`);
+        // if we dont get any events, covering any unexpected connection errors
+        throw new Error(`No events were found for ${dsnpUserId.toString()}`);
+      }
+
+      if(!this.blockchainService.api.events.utility.BatchCompleted.is(event)) {
+        this.logger.warn(`Batch failed event found for ${dsnpUserId.toString()}: ${event}`);
+        if(this.blockchainService.api.events.utility.BatchInterrupted.is(event)) {
+          // this event should not occur given we are filtering target event in extrinsic
+          // call to be `BatchCompleted`
+          this.logger.warn(`Unexpected event found for ${dsnpUserId.toString()}`);
+          this.logger.warn(event);
+          this.logger.warn(eventMap);
+          throw new Error(`Batch interrupted event found for ${dsnpUserId.toString()}`);
+        }
       }
+    } catch (e) {
+      this.logger.error(`Error processing batch for ${dsnpUserId.toString()}: ${e}`);
+      //Following errors includes are checked against
+      // 1. Inability to pay some fees`
+      // 2. Transaction is not valid due to `Target page hash does not match current page hash`
+
+      if (e instanceof Error && e.message.includes('Inability to pay some fees')) {
+        // in case capacity is low pause the queue
+        this.graphUpdateQueue.pause();
+        throw e;
+      } else if (e instanceof Error && e.message.includes('Target page hash does not match current page hash')) {
+        // refresh state and queue a non-transitive graph update
+        // this is safe to do as we are only updating single user's graph
+        const { key: jobId, data } = createGraphUpdateJob(dsnpUserId, provideId, SkipTransitiveGraphs);
+        this.graphUpdateQueue.add('graphUpdate', data, { jobId });
+        return;
```

Comment:
If page is stale, do non-transitive updates, any other errors are direct result of bad configuration so we need to only check for these two here